Tasks/Common/azure-arm-rest/azure-graph.ts

```diff
@@ -0,0 +1,77 @@
+import msRestAzure = require("./azure-arm-common");
+import azureServiceClient = require("./AzureServiceClient");
+import webClient = require("./webClient");
+import util = require("util");
+import Q = require("q");
+
+export class GraphManagementClient extends azureServiceClient.ServiceClient {
+    public servicePrincipals: ServicePrincipals;
+
+    constructor(credentials: msRestAzure.ApplicationTokenCredentials, subscriptionId: string, baseUri?: any, options?: any) {
+        super(credentials, subscriptionId);
+
+        this.apiVersion = '1.6';
+        this.acceptLanguage = 'en-US';
+        this.generateClientRequestId = true;
+
+        if (!options)
+            options = {};
+
+        if (baseUri) {
+            this.baseUri = baseUri;
+        }
+
+        if (options.acceptLanguage) {
+            this.acceptLanguage = options.acceptLanguage;
+        }
+        if (options.longRunningOperationRetryTimeout) {
+            this.longRunningOperationRetryTimeout = options.longRunningOperationRetryTimeout;
+        }
+        if (options.generateClientRequestId) {
+            this.generateClientRequestId = options.generateClientRequestId;
+        }
+
+        this.servicePrincipals = new ServicePrincipals(this);
+    }
+}
+
+export class ServicePrincipals {
+    private client: GraphManagementClient;
+
+    constructor(graphClient: GraphManagementClient) {
+        this.client = graphClient;
+    }
+
+    public async GetDefaultServicePrincipal(options): Promise<any> {
```

Comment:
Other appid's will fail with permission error. Exposing generic function can set wrong expectations for consumers. What i can change is to GetServicePrincipal() without appid argument. This makes it specific and open to future modifications as well. Later someone can add an optional parameter appid if it works. What do you say?